client/src/App.js

```diff
@@ -2,17 +2,51 @@ import { useState } from "react";
 import "./App.css";
 import DataList from "./components/DataList";
 import GetDataForm from "./components/GetDataForm";
+import LineGraph from "./components/Graph";
 
 function App() {
   const [rowData, setRowData] = useState([]);
+  const [allData, setAllData] = useState([]);
 
   const setError = (error) => {
     alert(error);
   };
 
+  const getRandomColor = () => {
+    var letters = "0123456789ABCDEF";
+    var color = "#";
+    for (var i = 0; i < 6; i++) {
+      color += letters[Math.floor(Math.random() * 16)];
+    }
+    return color;
+  };
+
+  const updateAllData = (data) => {
```

Comment:
this function isn't very readable.

add comments or refactor to not use so many operators in one line.

Reply:
as a general rule of thumb, do not use ternary operator ? : in complex expressions. In this case it would be better to use if statements. Evlis statements used for simple things like const x = expression ? 1 : 2